MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/AdsImaSDKListener.java

```diff
@@ -23,53 +26,71 @@ public AdsImaSDKListener(MuxBaseExoPlayer listener) {
 
     @Override
     public void onAdError(AdErrorEvent adErrorEvent) {
-        if (exoPlayerListener != null)
-            exoPlayerListener.dispatch(new com.mux.stats.sdk.core.events.playback.AdErrorEvent(null));
+        if (exoPlayerListener != null) {
+            PlaybackEvent event = new com.mux.stats.sdk.core.events.playback.AdErrorEvent(null);
+            setupAdViewData(event, null);
+            exoPlayerListener.dispatch(event);
+        }
+    }
+
+    private void setupAdViewData(PlaybackEvent event, Ad ad) {
+        ViewData viewData = new ViewData();
+        if (exoPlayerListener.getCurrentPosition() == 0) {
+            if (ad != null) {
+                viewData.setViewPrerollAdId(ad.getAdId());
+                viewData.setViewPrerollCreativeId(ad.getCreativeId());
+            }
+        }
+        event.setViewData(viewData);
     }
 
     @Override
     public void onAdEvent(AdEvent adEvent) {
         if (exoPlayerListener != null) {
+            PlaybackEvent event;
+            Ad ad = adEvent.getAd();
             switch (adEvent.getType()) {
-                case AD_BREAK_STARTED:
-                    exoPlayerListener.dispatch(new AdBreakStartEvent(null));
+                case CONTENT_PAUSE_REQUESTED:
+                    event = new AdBreakStartEvent(null);
+                    setupAdViewData(event, ad);
+                    exoPlayerListener.dispatch(event);
+                    event = new AdRequestEvent(null);
                     break;
-                case AD_BREAK_ENDED:
-                    exoPlayerListener.dispatch(new AdBreakEndEvent(null));
+                case CONTENT_RESUME_REQUESTED:
+                    event = new AdBreakEndEvent(null);
+                    break;
+                case LOADED:
+                    event = new AdResponseEvent(null);
```

Comment:
We should only send AdResponse if we know that there is a AdRequest in flight

Reply:
fixed